domain/application/state/storage.go

```diff
@@ -6,20 +6,40 @@ package state
 import (
 	"context"
 	"database/sql"
+	"fmt"
+	"path"
+	"strings"
 
 	"github.com/canonical/sqlair"
 	"github.com/juju/collections/set"
+	jujuerrors "github.com/juju/errors"
 
+	coreapplication "github.com/juju/juju/core/application"
+	"github.com/juju/juju/core/model"
+	"github.com/juju/juju/core/paths"
 	corestorage "github.com/juju/juju/core/storage"
 	coreunit "github.com/juju/juju/core/unit"
 	"github.com/juju/juju/domain/application"
+	applicationerrors "github.com/juju/juju/domain/application/errors"
+	"github.com/juju/juju/domain/life"
+	domainsequence "github.com/juju/juju/domain/sequence"
+	domainstorage "github.com/juju/juju/domain/storage"
 	storageerrors "github.com/juju/juju/domain/storage/errors"
+	"github.com/juju/juju/internal/charm"
 	"github.com/juju/juju/internal/errors"
 	"github.com/juju/juju/internal/storage"
 )
 
-// insertStorage constructs inserts storage directive records for the application.
-func (st *State) insertStorage(ctx context.Context, tx *sqlair.TX, appDetails applicationDetails, appStorage []application.AddApplicationStorageArg) error {
+// These consts are the sequence namespaces used to generate
+// monotonically increasing ints to use for storage entity IDs.
+const (
+	filesystemNamespace = "filesystem"
+	volumeNamespace     = "volume"
+	storageNamespace    = "storage"
+)
```

Comment:
I'd rather these be typed.

```suggestion
type storageSequenceNamespace string

const (
	filesystemNamespace storageSequenceNamespace = "filesystem"
	volumeNamespace storageSequenceNamespace  = "volume"
	storageNamespace  storageSequenceNamespace  = "storage"
)
```

Reply:
sure, i didn't do it because these are not exported and didn't feel it paid its way to do it

Reply:
And when we use it, eg `domainsequence.NextValue(ctx, st, tx, storageNamespace)` we'd only have to cast back to a string anyway so I can't see how it adds any value